server/http/controller.go

```diff
@@ -287,3 +288,27 @@ func (c *controller) UpdateTest(ctx context.Context, testID string, in openapi.T
 
 	return openapi.Response(204, nil), nil
 }
+
+func (c *controller) AssertionDryRun(ctx context.Context, _, runID string, def openapi.TestDefinition) (openapi.ImplResponse, error) {
+	rid, err := uuid.Parse(runID)
+	if err != nil {
+		return openapi.Response(http.StatusUnprocessableEntity, err.Error()), err
+	}
+
+	run, err := c.testDB.GetRun(ctx, rid)
+	if err != nil {
+		return openapi.Response(http.StatusInternalServerError, ""), nil
+	}
+
+	if run.Trace == nil {
+		return openapi.Response(http.StatusInternalServerError, "given run has no trace associated"), nil
```

Comment:
I think `StatusUnprocessableEntity` is best suitable here. We wouldn't want our API monitor alarms to go off because someone is trying to dry run something that didn't finish yet.